homeassistant/components/modbus/binary_sensor.py

```diff
@@ -109,33 +107,18 @@ def available(self) -> bool:
         """Return True if entity is available."""
         return self._available
 
-    def update(self):
+    async def async_update(self):
         """Update the state of the sensor."""
-        try:
-            if self._input_type == DEFAULT_INPUT_TYPE_COIL:
-                result = self._hub.read_coils(self._slave, self._address, 1)
-            else:
-                result = self._hub.read_discrete_inputs(self._slave, self._address, 1)
-        except ConnectionException:
-            self._set_unavailable()
+        if self._input_type == CALL_TYPE_COIL:
+            result = await self._hub.read_coils(self._slave, self._address, 1)
+        else:
+            result = await self._hub.read_discrete_inputs(self._slave, self._address, 1)
+        if result is None:
+            self._available = False
             return
-
         if isinstance(result, (ModbusException, ExceptionResponse)):
-            self._set_unavailable()
+            self._available = False
             return
 
         self._value = result.bits[0]
         self._available = True
-
-    def _set_unavailable(self):
```

Comment:
Ideally this retain the logging when it goes unavailable per https://developers.home-assistant.io/docs/integration_quality_scale_index/

"Handles device/service unavailable. Log a warning once when unavailable, log once when reconnected."

Reply:
Confusion, I think you are looking at an old push. _set_unavailable() no longer exist in the code. We should only log once for every problem.